### PR TITLE
Add support for symfony 6

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: .github/workflows/test-application.yaml [hunk 1]
```diff
@@ -16,13 +16,7 @@ jobs:
...
             matrix:
                 include:
-                    - php-version: '5.6'
-                      elasticsearch-version: '5.6.14'
-                      lint: false
-                      symfony-version: '^2.8'
-                      elasticsearch-package-constraint: '^5.0'
-
-                    - php-version: '7.0'
+                    - php-version: '7.2'
```
Comment (niklasnatter) [inline .github/workflows/test-application.yaml:19]: the new version of the `symfony/phpunit-bridge` does not include the `SetUpTearDownTrait` anymore. therefore we cannot support php versions that are compatible with php@5.6

File: Service/ManagerFactory.php [hunk 2]
```diff
@@ -169,7 +170,7 @@ public function createManager($managerName, $connection, $analysis, $managerConf
 
     private function dispatch($eventName, $event)
     {
-        if (class_exists(LegacyEventDispatcherProxy::class)) {
+        if ($this->eventDispatcher instanceOf ContractsEventDispatcherInterface || class_exists(LegacyEventDispatcherProxy::class)) {
```
Comment (niklasnatter) [inline Service/ManagerFactory.php]: symfony 5.4 does not include the `LegacyEventDispatcherProxy` anymore

File: .github/workflows/test-application.yaml (hunk 1 above)
Comment (alexander-schranz) [reply .github/workflows/test-application.yaml:19]: we should drop then php@5.6 from the composer.json

Comment (niklasnatter) [reply .github/workflows/test-application.yaml:19]: done 🙂

File: Service/Manager.php [hunk 3]
```diff
@@ -734,7 +735,7 @@ private function stopwatch($action, $name)
 
     private function dispatch($eventName, $event)
     {
-        if (class_exists(LegacyEventDispatcherProxy::class)) {
+        if ($this->eventDispatcher instanceOf ContractsEventDispatcherInterface || class_exists(LegacyEventDispatcherProxy::class)) {
```
Comment (alexander-schranz) [inline Service/Manager.php]:
```suggestion
        if ($this->eventDispatcher instanceof ContractsEventDispatcherInterface || class_exists(LegacyEventDispatcherProxy::class)) {
```

File: Service/ManagerFactory.php (hunk 2 above)
Comment (alexander-schranz) [inline Service/ManagerFactory.php]:
```suggestion
        if ($this->eventDispatcher instanceof ContractsEventDispatcherInterface || class_exists(LegacyEventDispatcherProxy::class)) {
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**